src/operations/blas3/gemm_common.hpp

```diff
@@ -42,12 +42,22 @@ mul_add(T a, T b, T c,
 }
 #endif
 
+#ifndef BLAS_ENABLE_HALF
 template <typename T>
 static PORTBLAS_INLINE T
 mul_add(T a, T b, T c,
         typename std::enable_if<is_sycl_scalar<T>::value>::type * = 0) {
   return (cl::sycl::mad(a, b, c));
 }
+#else
+template <typename T>
```

Comment:
If you remove the `BLAS_ENABLE_HALF` guard from `is_half` in `include/blas_meta.h`, you can also remove this `#ifndef` pragma keeping just the `false` case.